src/const.js

```diff
@@ -121,12 +122,39 @@ export const MIME_TO_FILE_EXTENSIONS = {
 // List of the mime types for the allowed static theme images.
 export const STATIC_THEME_IMAGE_MIMES = Object.keys(MIME_TO_FILE_EXTENSIONS);
 
-// List of the "schema data paths" of the deprecated static theme's LWT aliases.
-export const DEPRECATED_STATIC_THEME_LWT_ALIASES = [
-  '/theme/images/headerURL',
-  '/theme/colors/accentcolor',
-  '/theme/colors/textcolor',
-];
+// Mapping of "schema data paths" of the deprecated properties that we
+// issue warnings for.
+// If the value is `null` we will be using the `deprecated` message
+// from the schema. Otherwise `code`, `message` and `description` will be taken
+// from the object provided.
+export const DEPRECATED_MANIFEST_PROPERTIES = {
+  '/theme/images/headerURL': 'MANIFEST_THEME_LWT_ALIAS',
+  '/theme/colors/accentcolor': 'MANIFEST_THEME_LWT_ALIAS',
+  '/theme/colors/textcolor': 'MANIFEST_THEME_LWT_ALIAS',
+};
+
+// Mapping of deprecated javascript apis.
+// If the value is `null` we will be using the `deprecated` message
+// from the schema. Otherwise `code`, `message` and `description` will be taken
+// from the object provided.
+export const DEPRECATED_JAVASCRIPT_APIS = {
+  // These APIs were already deprecated by Chrome and Firefox never
+  // supported them. We do still issue deprecation warnings for them.
+  'app.getDetails': 'DEPRECATED_CHROME_API',
+  'extension.onRequest': 'DEPRECATED_CHROME_API',
+  'extension.onRequestExternal': 'DEPRECATED_CHROME_API',
+  'extension.sendRequest': 'DEPRECATED_CHROME_API',
+  'tabs.getAllInWindow': 'DEPRECATED_CHROME_API',
+  'tabs.getSelected': 'DEPRECATED_CHROME_API',
+  'tabs.onActiveChanged': 'DEPRECATED_CHROME_API',
+  'tabs.onSelectionChanged': 'DEPRECATED_CHROME_API',
+  'tabs.sendRequest': 'DEPRECATED_CHROME_API',
+
+  // https://github.com/mozilla/addons-linter/issues/2556
+  'proxy.register': null,
```

Comment:
Nit, deprecated APIs with null as the value are going to get 'DEPRECATED_API' as the code assigned by default, right? we may set it explicitly (only to make it immediately visible, the actual behavior should not be affected by specify it explicitly vs. leave it as implicit as it is right now).

Reply:
Great idea, added in b6fe8ead